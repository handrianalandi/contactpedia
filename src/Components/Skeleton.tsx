import { keyframes } from "@emotion/react";
import styled from "@emotion/styled";
import React from "react";

const loadingAnimation = keyframes`
0% {
    background-color: hsl(200, 20%, 80%);
  }
  100% {
    background-color: hsl(200, 20%, 95%);
  }
`;
const SkeletonElement = styled.div`
  width: 100%;
  border-radius: 15px;
  margin-bottom: 10px;
  animation: ${loadingAnimation} 1s linear infinite alternate;
`;

const SkeletonCard = styled(SkeletonElement)`
  height: 50px;
`;

const SkeletonTextLong = styled(SkeletonElement)`
  height: 20px;
  width: 80%;
`;

const SkeletonTextShort = styled(SkeletonElement)`
  height: 20px;
  width: 40%;
`;

export default function Skeleton() {
  return (
    <>
      <SkeletonCard />
      <SkeletonTextLong />
      <SkeletonTextShort />
    </>
  );
}