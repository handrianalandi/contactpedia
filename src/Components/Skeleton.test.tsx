import {render} from '@testing-library/react';
import Skeleton from './Skeleton';
describe('Skeleton', () => {
    it('should render correctly', () => {
        const {container} = render(<Skeleton/>);
        expect(container).toMatchSnapshot();
    }
    );
});