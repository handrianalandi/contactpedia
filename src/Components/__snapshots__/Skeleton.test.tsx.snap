// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render correctly 1`] = `
<div>
  <div
    class="css-c62n0o"
  />
  <div
    class="css-dbrq19"
  />
  <div
    class="css-7hekhs"
  />
</div>
`;
